src/GovUk.Education.ExploreEducationStatistics.Data.Services/TimePeriodService.cs

```diff
@@ -48,6 +50,26 @@ public IEnumerable<(int Year, TimeIdentifier TimeIdentifier)> GetTimePeriods(Gui
             return TimePeriodUtil.GetTimePeriodRange(start, end);
         }
 
+        public TimePeriodRangeLabels GetTimePeriodRangeLabels(Guid subjectId)
+        {
+            var observationsQuery = _context
```

Comment:
Bear in mind that there could be hundreds of thousands of these, and pulling them all out of the db might be slow.  Maybe try testing using an nd01_anon file, which hs around 200,000 rows?

Reply:
I do see though that this hooks into pre-existing code, but I think if there's an inefficiency here, it'll strike in the first line of `GetDistinctObservationTimePeriods()` that this code calls.

Reply:
My bad, sorry!